modules/db/src/main/java/org/mule/module/db/internal/domain/param/DefaultInputQueryParam.java

```diff
@@ -14,20 +14,35 @@
 
     private final Object value;
 
+    private final boolean hasValue;
+
     public DefaultInputQueryParam(int index, DbType type, Object value)
     {
         this(index, type, value, null);
     }
 
     public DefaultInputQueryParam(int index, DbType type, Object value, String name)
     {
+        this(index, type, value, name, value != null);
+    }
+
+    private DefaultInputQueryParam(int index, DbType type, Object value, String name, boolean hasValue)
```

Comment:
I don't understand why do you need this private constructor. In a previous version of your code, you were calling this, in that that time, public constructor with different boolean values. But now, there is no need to have it as it is used only from one constructor.
In a previous comment, the suggestion was to add a constructor without the "value" parameter, so the caller could invoke different ones depending on the given situation, and it that scenario a private constructor was useful. But now I'm lost